record-and-playback/core/scripts/rap-events-worker.rb

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/ruby
+# encoding: UTF-8
+
+# Copyright ⓒ 2017 BigBlueButton Inc. and by respective authors.
+#
+# This file is part of BigBlueButton open source conferencing system.
+#
+# BigBlueButton is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version.
+#
+# BigBlueButton is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+# 
+# You should have received a copy of the GNU Lesser General Public License
+# along with BigBlueButton.  If not, see <http://www.gnu.org/licenses/>.
+
+require '../lib/recordandplayback'
+require 'rubygems'
+require 'yaml'
+require 'fileutils'
+
+def run_post_events_script(meeting_id)
+	Dir.glob("post_events/*.rb").sort.each do |post_event_script|
+		match = /([^\/]*).rb$/.match(post_event_script)
+		post_type = match[1]
+		BigBlueButton.logger.info("Running post event script #{post_type}")
+
+		step_start_time = BigBlueButton.monotonic_clock
+		ret = BigBlueButton.exec_ret("ruby", post_event_script, "-m", meeting_id)
+		step_stop_time = BigBlueButton.monotonic_clock
+		step_time = step_stop_time - step_start_time
+		step_succeeded = (ret == 0)
+
+		if not step_succeeded
+			BigBlueButton.logger.warn("Post event script #{post_event_script} failed")
+		end
+	end
+end
+
+def keep_events_from_non_recorded_meeting(recording_dir, ended_done_file)
```

Comment:
Is it for only non-recorded meetings?

Reply:
The `rap-events-worker.rb` is for both recorded and non-recorded meetings. The `rap-archive-worker.rb` doesn't do anything related to keeping events.

Reply:
Ok. Then can we change the name of this method for something like `keep_events_from_meeting`?

Reply:
Ok. Renamed it to `keep_meeting_events`.